src/Simple.OData.Client.Core/Fluent/FluentCommand.cs

```diff
@@ -258,6 +260,31 @@ public FluentCommand Key(IDictionary<string, object> key)
 
             _details.NamedKeyValues = key;
             _details.KeyValues = null;
+            _details.IsAlternateKey = false;
+            return this;
+        }
+
+        public FluentCommand AltKey(object key)
+        {
+            if (IsBatchResponse) return this;
+
+            if (key != null && TypeCache.IsAnonymousType(key.GetType()))
+            {
+                _details.NamedKeyValues = TypeCache.ToDictionary(key);
+            }
```

Comment:
This should probably raise an argument exception if an object is used that cannot be made into a `NamedKeyValues` dictionary. As this stands it would also not support `client.AltKey("name")` looking up that there is only a single alternate key; an exception should be raised in that instance too.

Reply:
That was my initial thought but reading through the code (for that existing Key() implementation) I noticed that
- No validation is performed when setting the values using Key()
- Validation is only performed when building the CommandText at which point the only thing done is filtering out invalid names (if named pairs given) or zipping together the values (if values given)
- Alternate Keys are not handled in the metadata

So I followed that pattern. If you want I can perform another PR that handles Alternate Key meta and performs actual validation on both Primary or Alternate keys.  For alternate keys I could also make the discovery of FilterAsKey working for alternate keys.

Reply:
If there is not any validation along the `Key` path, I'm okay with this as-is for this PR. Certainly a subsequent PR could include the validation.

Reply:
In that case, now that I think about it a separate AltKey method would be obsolete. OData it self has one syntax for primary and alternate key so let's follow that. Let me follow up on this PR and consume the alternate key metadata to have the Key method handle everything.